### ISSUE TITLE
Support instantiation with 'adata=None' + LightningDataModule for training

### ISSUE THREAD

**Issue by xiaotong743** [theislab/DRVI#30]
### Description of feature

Hi Amir,

Thanks for the amazing work!

We’re training scVI on large datasets that require approximate shuffling, which isn’t supported by AnnData. In scVI, there is an EXPERIMENTAL feature where user can instantiate without adata and pass the datamodule during training:
```
model = scvi.model.SCVI(adata=None)
model.train(datamodule=datamodule)
```
It would be great to have the same support in DRVI. Thanks again.

**Comment by moinfar** [comment]
Hi,

Thanks!

Thanks for the suggestion. This technically depends on Issue #22, as the `datamodule` was recently added to scvi-tools (DRVI depends on an older version).
I will inform you and ask for your help with testing it when it is done.

**Comment by moinfar** [comment]
@xiaotong743 
I have created #33 to solve this problem.
Can you please check the `update-scvi-version` branch? In theory, it should be possible to do the same in this branch.

**Comment by xiaotong743** [comment]
@moinfar 
Right now, it still returns:
'DRVI.__init__() missing 1 required positional argument: 'adata''. 
It looks like the 'BaseModelClass' in scVI explicitly allows only GIMVI and SCVI to be instantiated without adata or a registry by a condition:
```
elif (self.__class__.__name__ == "GIMVI") or (self.__class__.__name__ == "SCVI"):
            # note some models do accept empty registry/adata (e.g: gimvi)
            pass
```

**Comment by moinfar** [comment]
@xiaotong743 
Do you have any registry? What are the covariate information?

**Comment by moinfar** [comment]
In addition to my previous question, I took a look at the scvi-tools experimental changes. The new code makes the extension of scVI much more complicated and, in some cases, impossible. I prefer to defer any further alignment in this direction as I am sure some changes will be reverted.

I have two suggestion:

1. For now, I made passing adata=None but passing registry possible.
I think this way, you should be able to train the model using your custom loader. Please check [here](https://github.com/scverse/scvi-tools/pull/2932#issuecomment-2843497052). You should hopefully be able to get a proper registry dictionary using a dummy anndata and initialize the model with that.

2. Alternatively, I think maybe it was better if scvi-tools didn't make that much complication, and a user could initialize the model using an empty anndata with same structure of obs and var:
```
n_cells = 1_000_000
n_genes = 4000

# Create a sparse CSR matrix with all zeros
X = sp.csr_matrix((n_cells, n_genes), dtype=np.float32)

# Create the AnnData object
adata = ad.AnnData(X)
adata.obs = ...
adata.var = ...
```
Then the user [theoretically should be able to](https://github.com/scverse/scvi-tools/blob/main/src/scvi/model/base/_training_mixin.py#L123-L143) pass a compatible datamodule to model.train(...).

**Comment by moinfar** [comment]
Closed since we now have a more general issue. #22
